### PR TITLE
fix(tree): allow single select only and add indicator

### PR DESCRIPTION
**Pull request by josercarcamo**
**Related Issue:** #7899

## Summary
Allowed only single selection and added the selection indicator

### CODE REVIEW

File: packages/calcite-components/src/components/tree-item/tree-item.tsx [hunk 1]
```diff
@@ -201,7 +201,10 @@ export class TreeItem implements ConditionalSlotComponent, InteractiveComponent
 
   render(): VNode {
     const rtl = getElementDir(this.el) === "rtl";
-    const showBulletPoint = this.selectionMode === "single" || this.selectionMode === "children";
```
Comment (driskull) [inline packages/calcite-components/src/components/tree-item/tree-item.tsx:204]: Do existing screenshot tests show changes? If not, can we add a screenshot test for he bullet point?

File: packages/calcite-components/src/components/tree/tree.tsx [hunk 2]
```diff
@@ -175,7 +175,8 @@ export class Tree {
       (((this.selectionMode === "single" || this.selectionMode === "multiple") &&
         childItems.length <= 0) ||
         this.selectionMode === "children" ||
-        this.selectionMode === "multichildren");
+        this.selectionMode === "multichildren" ||
+        this.selectionMode === "single-persist");
```
Comment (driskull) [inline packages/calcite-components/src/components/tree/tree.tsx:179]: Should we add an e2e test to account for single-persist allowing to be unselected?

**Review by driskull — COMMENTED** [review]
Hey @josercarcamo code changes look 💣 👍 

Just had some comments

File: packages/calcite-components/src/components/tree/tree.tsx (hunk 2 above)
Comment (josercarcamo) [reply packages/calcite-components/src/components/tree/tree.tsx:179]: Done.

**Review by driskull — APPROVED** [review]
👍 💯

**Pull request merged**